src/mococrw/asymmetric_encryption.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include <type_traits>
+
+#include "key.h"
+#include "openssl_wrap.h"
+#include "padding_mode.h"
+
+namespace mococrw {
+
+/**
+ * This class is used to perform RSA asymmetric encryption and decryption
+ */
+class AsymmetricEncryption
+{
+public:
+    AsymmetricEncryption() = delete;
+
+private:
+    std::vector<uint8_t> message;
+
+public:
+    /**
+     * @brief CryptoData
+     * 
+     * This helper class defines the input and output data type used for encryption and decryption.
+     * Currently this class abstracts two underlying types (std::string and std::vector<unit_8t>).
+     * The user is free to call the encrypt and decrypt functionalities with either of the available
+     * underlying types.
+     */
+    class CryptoData
```

Comment:
I like the idea of having reach collection of types. I wounder if we should use something like CryptoData widely across the library.